handler.go

```diff
@@ -37,12 +37,39 @@ type rpcHandler struct {
 
 type request struct {
 	Jsonrpc string            `json:"jsonrpc"`
-	ID      *int64            `json:"id,omitempty"`
+	ID      requestID         `json:"id,omitempty"`
 	Method  string            `json:"method"`
 	Params  []param           `json:"params"`
 	Meta    map[string]string `json:"meta,omitempty"`
 }
 
+type requestID struct {
+	actual interface{} // nil, int64, or string
+}
+
+func (r *requestID) UnmarshalJSON(data []byte) error {
+	switch data[0] {
+	case 'n': // null
+	case '"': // string
+		var s string
+		if err := json.Unmarshal(data, &s); err != nil {
+			return err
+		}
+		r.actual = s
+	default: // number
+		var n int64
+		if err := json.Unmarshal(data, &n); err != nil {
+			return err
+		}
+		r.actual = n
+	}
+	return nil
+}
+
+func (r requestID) MarshalJSON() ([]byte, error) {
```

Comment:
You could add a sanity check here that "actual" is one of the three allowed Go types, just as an extra sanity check since UarshalJSON does it too.

Reply:
thanks @mvdan, added and it picked up a bug in my code! can you sanity check my latest commit to see if it's how you would have done it?

Reply:
SGTM. You could simplify it a little bit, since you don't need to separate the json.Marshal calls:

```
switch r.actual.(type) {
	case nil, int64, string:
		return json.Marshal(r.actual)
	default:
		return nil, fmt.Errorf("unexpected ID type: %T", r.actual)
	}
```

Reply:
:thumbsup: thanks & done